ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java

```diff
@@ -643,6 +648,40 @@ private void printQueryDescription(final QueryDescription query) {
     printOverriddenProperties(query);
   }
 
+  private void printConnectorDescription(final ConnectorDescription description) {
+    final ConnectorStateInfo status = description.getStatus();
+    writer().println(String.format("%-20s : %s", "Name", status.name()));
+    writer().println(String.format("%-20s : %s", "Type", description.getType()));
+    writer().println(String.format("%-20s : %s", "State", status.connector().state()));
+    writer().println(String.format("%-20s : %s", "WorkerId", status.connector().workerId()));
+    writer().println();
+
+    if (!status.tasks().isEmpty()) {
+      final Table taskTable = new Table.Builder()
+          .withColumnHeaders(ImmutableList.of("Task ID", "State", "Error Trace"))
+          .withRows(status.tasks()
+              .stream()
+              .map(task -> ImmutableList.of(
+                  String.valueOf(task.id()),
+                  task.state(),
+                  ObjectUtils.defaultIfNull(task.trace(), ""))))
+          .build();
+      taskTable.print(this);
+      writer().println();
+    }
+
+    if (!description.getSources().isEmpty()) {
```

Comment:
What about SINK and UNKNOWN? Do these come in a different patch?

Also, I assume that even though we have separate CREATE SOURCE CONNECTOR xxx and CREATE SINK CONNECTOR xxx statements that the namespace for both are the same for the status query to work unambiguously?

Reply:
Terminology here is a little confusing. Internally, "source" refers to anything that is a Stream or a Table, not necessarily something that produces data (e.g. a source connector). So `description.getSources()` returns everything, including sinks.

Reply:
> Also, I assume that even though we have separate CREATE SOURCE CONNECTOR xxx and CREATE SINK CONNECTOR xxx statements that the namespace for both are the same for the status query to work unambiguously?

Yes. Later we may want to introduce specific behavior for each of them.

Reply:
Speaking from a naive user's perspective (which I am right now :)), I'm a bit confused on what's being displayed. Maybe a doc update with this patch would make it clearer?

For example, what is a task in this context? Is it the thing sucking data out or pushing data into a DB? Is it a StreamTask? Something else?

Is "source" the stream / table that contains the CDC data for sources? Is it the stream / table used to feed data into the sink database? Source might be a bit overloaded because we're using it as the syntax in the "CREATE SOURCE CONNECTOR" statement, but I think source means something different here?

Reply:
BTW, if this is already documented somewhere or the user is expected to understand the distinctions having gone through setting this up then I'm fine closing this out. I'm coming at this having only briefly interacted with connectors.

Reply:
It's definitely confusing because this is the point of intersection between connect and KafkaStreams, both of which have pretty overlapping sets of terminologies.

- **sources** refer to the KSQL concept of a source, which is any Stream or Table irrespective of whether data is being written into or read from
- **type** refers to the connect concept of `ConnectorType`, which can be `SOURCE`, `SINK` or `UNKNOWN` and this refers to whether the connector is moving data into or out of kafka
- **task** refers to a a connect worker task (https://docs.confluent.io/current/connect/concepts.html#connect-tasks) - so your intuition about it being the thing that sucks data out or pushes data into is correct (it is not a StreamTask, though it is in essence actually very similar)

So I think, correct me if I'm wrong, the confusion here is that while "type" and "task" are both connect concepts in the context, "sources" is actually using the KSQL concept. Perhaps it clarifies things if the title for that portion was `KSQL Sources`.

Let me know what you think, and thanks for the detailed thoughts!

Reply:
I think I'm even more confused on type now :). In your example in the test code below I see a top level type, which is the connector class and I see a type in the sources table, which appears to be STREAM or TABLE. I'm not seeing a type that indicates SOURCE, SINK, UNKNOWN.

I think you're right that the KSQL use of "source" is confusing, and that goes beyond the scope of this commit, so I don't really want to hold this up. It's probably fine to rename this KSQL Sources for now and perhaps revisit the name "source" in the context of KSQL if I'm not the only one confused :).

BTW, an example of how a source and a sink connector would be described might be helpful as a way to clear up some of my lingering confusion. If you have an example handy for both, would you mind sending them my way?

Reply:
Ah the example in the testing done section is out of date... I actually confused myself reading it! Here is a more recent one (and a sample of how a SINK connector would look like):
```
ksql> DESCRIBE CONNECTOR "jdbc-connector3";

Name                 : jdbc-connector3
Class                : io.confluent.connect.jdbc.JdbcSourceConnector
Type                 : source
State                : RUNNING
WorkerId             : 10.200.7.69:8083

 Task ID | State   | Error Trace
---------------------------------
 0       | RUNNING |
---------------------------------

 KSQL Source Name      | Kafka Topic | Type
---------------------------------------------
 JDBC_CONNECTOR3_USERS | jdbc2-users | TABLE
---------------------------------------------

ksql> DESCRIBE CONNECTOR "elsatic-sink";

Name                 : elastic-sink
Class                : io.confluent.connect.elasticsearch.ElasticsearchSinkConnector
Type                 : sink
State                : RUNNING
WorkerId             : 10.200.7.69:8083

 Task ID | State   | Error Trace
---------------------------------
 0       | RUNNING |
---------------------------------
```
The sink connectors are not yet implemented, but it'll look something like that (perhaps we can even show the KSQL sources that it reads from)

Reply:
Originally the type that indicated `SOURCE|SINK|UNKNOWN` was hidden from this view (it was present in the JSON response, but not the tabular one)